### PR TITLE
[RISC-V] Fix JitDisasm in release build

### PR DESCRIPTION
**Pull request by Bajtazar**
Fixes JitDisasm in release build which wasn't showing the assembly instructions

note: clang-format changed some indentations across the files

Part of #84834
cc @wscho77 @HJLeee @clamp03 @JongHeonChoi @t-mustafin @gbalykov @viewizard @ashaurtaev @sirntar @yurai007 @tomeksowi @brucehoult

### CODE REVIEW

**Comment by Bajtazar** [comment]
@dotnet-policy-service agree company="Samsung"

File: src/coreclr/jit/ee_il_dll.cpp [hunk 1]
```diff
@@ -144,12 +144,13 @@ FILE* jitstdout()
 }
 
 // Like printf/logf, but only outputs to jitstdout -- skips call back into EE.
-void jitprintf(const char* fmt, ...)
+int jitprintf(const char* fmt, ...)
```
Comment (jkotas) [inline src/coreclr/jit/ee_il_dll.cpp:147]: Why is this change needed?

Comment (Bajtazar) [reply src/coreclr/jit/ee_il_dll.cpp:147]: Inside the `emitDispInsName` in the `emitriscv64.cpp` file the return value of the `printf` function has been used in [some places](https://github.com/dotnet/runtime/blob/413efe748ac4c5daabf087ed8ed0d59c511e5cf2/src/coreclr/jit/emitriscv64.cpp#L3252). When I've allowed this function to be compiled in the release mode I've found out that the code wasn't compiling due to the void type returned by the `printf`. Upon inspection I thought that changing signature of this function would be the easiest fix

File: src/coreclr/jit/emitriscv64.cpp [hunk 2]
```diff
@@ -2854,7 +2855,12 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             assert(!"JitBreakEmitOutputInstr reached");
         }
     }
-#endif
+#else  // !DEBUG
+    if (emitComp->opts.disAsm)
+    {
+        emitDispIns(id, false, false, true, 0, *dp, (dstRW - odstRW), ig);
+    }
+#endif // !DEBUG
```
Comment (tomeksowi) [inline src/coreclr/jit/emitriscv64.cpp:2863]: Nit: the comment for #endif should reflect the #if, not #else condition.

Ditto elsewhere.

**Review by tomeksowi — CHANGES_REQUESTED** [review]

File: src/coreclr/jit/emitriscv64.cpp [hunk 3]
```diff
@@ -1981,8 +1981,8 @@ void emitter::emitJumpDistBind()
                 instruction ins = jmp->idIns();
                 assert((INS_jal <= ins) && (ins <= INS_bgeu));
 
-                if (ins > INS_jalr ||
-                    (ins < INS_jalr && ins > INS_j)) // jal < beqz < bnez < jalr < beq/bne/blt/bltu/bge/bgeu
+                if (ins > INS_jalr || (ins < INS_jalr && ins > INS_j)) // jal < beqz < bnez < jalr <
+                                                                       // beq/bne/blt/bltu/bge/bgeu
```
[the remark this answers is not in the document]
Comment (Bajtazar) [reply src/coreclr/jit/emitriscv64.cpp]: I've used the invalid version of the clang-format. Is is fixed now

File: src/coreclr/jit/emitriscv64.cpp [hunk 4]
```diff
@@ -2854,7 +2847,12 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             assert(!"JitBreakEmitOutputInstr reached");
         }
     }
-#endif
+#else  // !DEBUG
+    if (emitComp->opts.disAsm)
+    {
+        emitDispIns(id, false, false, true, 0, *dp, (dstRW - odstRW), ig);
+    }
+#endif // !DEBUG
```
Comment (tomeksowi) [inline src/coreclr/jit/emitriscv64.cpp:2855]: Why can't you choose whether to display offsets (emitComp->opts.disDiffable) like in debug?

Comment (Bajtazar) [reply src/coreclr/jit/emitriscv64.cpp:2855]: Both [x86](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/emitxarch.cpp#L17515) and [arm64](https://github.com/dotnet/runtime/blob/88b5e3d4b77dd8238331ade1b31ac8ddc62f22f7/src/coreclr/jit/emitarm64.cpp#L14361) doesn't allow user to display offsets in release mode. I wanted my implementation to behave similarly to those two

**Review by tomeksowi — APPROVED** [review]

File: src/coreclr/jit/emitriscv64.cpp [hunk 5]
```diff
@@ -3829,12 +3816,23 @@ void emitter::emitDispInsHex(instrDesc* id, BYTE* code, size_t sz)
 void emitter::emitDispIns(
     instrDesc* id, bool isNew, bool doffs, bool asmfm, unsigned offset, BYTE* pCode, size_t sz, insGroup* ig)
 {
-    // RISCV64 implements this similar by `emitter::emitDisInsName`.
-    // For RISCV64 maybe the `emitDispIns` is over complicate.
-    // The `emitter::emitDisInsName` is focused on the most important for debugging.
-    NYI_RISCV64("RISCV64 not used the emitter::emitDispIns");
+    if (pCode == nullptr)
```
Comment (clamp03) [inline src/coreclr/jit/emitriscv64.cpp:3819]: It is good. However, how about displaying all instructions of methods in `a new function` or `caller function`? 
As I know, `emitDispIns` is too complicate for LOONGARCH64. LOONGARCH64 implemented a new function `emitDisInsName` instead of `emitDispIns`. And I just follow LOONGARCH64 for RISCV64. So RISCV64's Implementation is too different from the other archs's. Moreover, `emitDispIns` displays single instruction in the other archs.

Comment (Bajtazar) [reply src/coreclr/jit/emitriscv64.cpp:3819]: I agree that the `emitDispIns` function has too much utilities and ultimately can be viewed as an overkill for the RISC64's platform. On the other hand all of the other architectures (including [LoongArch64](https://github.com/dotnet/runtime/blob/6249fd2e59ed1308eb7cf976fb48c1ecab5eaa4e/src/coreclr/jit/emitloongarch64.cpp#L5958)) has implemented this function (although it seems that the LA64 doesn't use it anywhere which is strange considering the function code resemblance to some other parts of their code like [here](https://github.com/dotnet/runtime/blob/6249fd2e59ed1308eb7cf976fb48c1ecab5eaa4e/src/coreclr/jit/emitloongarch64.cpp#L3774C1-L3779C6)). In my opinion it would be more beneficial to discard some of the values passed to this function and use it in the simplified form since the new reader accustomed to other archs implementation would have much better time understanding the RISC64 implementation. I agree that `emitDispIns` display only one instruction on the other archs but it also seems that on those archs the `emitOutputInstr` emits only one instruction per call which is not the case for the RISC64's implementation. I also believe that allowing this function to exist with behavior similar to the other archs will allow to decrease the number of `ifdef`'s in the `emit.cpp` file which will result with more generalized and easier to understand code

**Review by sirntar — APPROVED** [review]

**Review by ashaurtaev — APPROVED** [review]

**Comment by Bajtazar** [comment]
@jakobbotsch Could you review this PR? Thanks in advance

File: src/coreclr/jit/emitriscv64.cpp (hunk 5 above)
Comment (jakobbotsch) [reply src/coreclr/jit/emitriscv64.cpp:3819]: > I also believe that allowing this function to exist with behavior similar to the other archs will allow to decrease the number of `ifdef`'s in the `emit.cpp` file which will result with more generalized and easier to understand code

:+1: -- this should always be the goal, anything else makes maintenance and development very difficult for us. The backends should behavior as similarly as possible.

Comment (jakobbotsch) [reply src/coreclr/jit/emitriscv64.cpp:3819]: Can you please fix this so that `emitDispIns` behaves as other platforms as so that this ifdef can be removed:

https://github.com/dotnet/runtime/blob/ea13b5bd2c17e970590068d9dfce8171b663cde1/src/coreclr/jit/emit.cpp#L4197-L4198

**Review by clamp03 — APPROVED** [review]

File: src/coreclr/jit/emitriscv64.cpp (hunk 5 above)
Comment (Bajtazar) [reply src/coreclr/jit/emitriscv64.cpp:3832]: Ok, done

**Review by jakobbotsch — APPROVED** [review]

**Comment by tomeksowi** [comment]
Congratulations on landing your first PR:)
